docs/proposals/status.md

```diff
@@ -0,0 +1,300 @@
+- Simplify the status tracking requirements for a consumer of Application CRD. 
+- Status aggregation should aim to reduce the entropy of information that the consumers of Application CRD's `.status` would otherwise have to deal with. If a consumer needs a more detailed status of the underlying components, they could query the underlying components status.
+- Propose a minimal set of standardized Conditions for all resources to implement.
+
+## Scope
+The scope is limited to the objects matching the application crd selectors. The scope is also limited to using the information present in the `.status` field for kubernetes objects. External systems to augment health (black/white box) is not part of this scope.
+
+## Proposal
+The proposal is to add a reconciler to the Application CRD controller, that creates watches for the GroupKinds that the application CRD refers to. The reconciler then inspects the matching objects and updates the aggregate status in the Application CRD. Lowering information entropy and reducing the amount of interpertation to be done by the consumers of status is important for the adoption.
+
+### Fields
+Fields are a structured way to capture the status of resources. As part of this proposal, no standardized fields are proposed. The authors opinion is it does not provide additional information than what conditions provide. If the community strongly feels about standardizing some fields for in-tree and custom resources, it can be considered. 
+
+### Conditions
+[Conditions](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#typical-status-properties) are used to describe the status of resources in an extensible way. Out of tree controllers support a variety of conditions. There is no consistent usage across the core controller. This is exacerbated with custom controllers. To tackle this diversity and its impact in status computation for Application CRD, we need a short term and a long term solution. Short term solution is to do aggregation of diverse core resources in the Application CRD controller. Long term is to identify best practices and standardize conditions and propagate them to the core controllers thereby making Application CRD aggregation simpler.
+
+#### Standard Conditions
+These conditions should be added to `.status.conditions` for Application CRD and subsequently to other resources:
+##### `Ready`
+Readiness as perceived by the resource controller. This means the controller deems the underlying resources to be ready for consumption to the best of its knowledge. It coule be implemented as a low pass filter to filter out transient or persistent failures such as restarts of underlying pods.
+
+##### `Settled`
+Would indicate whether controller acted on latest revision of `.spec` and the underlying resources conform to the latest revision of the `.spec`
+
+##### `Error`
+Would capture Errors seen in last reconcilation loop.
```

Comment:
Which errors would be reported here? The proposal seems to suggest it would be "shallow" errors (_e.g._, "this particular `Deployment` failed to progress"), rather than the transitive closure of all errors in all resources managed by the app. CRD (_e.g._, "here are all the `Pod` crash errors that caused this particular `Deployment` to fail")?

Reply:
It depends on the controller design. We can make recommendations on how the error needs to be propagated up the hierarchy. But iam sure some controllers may end up hiding the errors. 
Its a good point and we can explicitly keep it as a design goal when propagating this condition to all core controllers.